src/FEValues/interface_values.jl

```diff
@@ -0,0 +1,372 @@
+  It defaults to `func_interpol_b`.
+ 
+**associated methods:**
+
+* [`shape_value_average`](@ref)
+* [`shape_value_jump`](@ref)
+* [`shape_gradient_average`](@ref)
+* [`shape_gradient_jump`](@ref)
+
+**Common methods:**
+
+* [`reinit!`](@ref)
+* [`getnquadpoints`](@ref)
+* [`getdetJdV`](@ref)
+
+* [`shape_value`](@ref)
+* [`shape_gradient`](@ref)
+* [`shape_divergence`](@ref)
+* [`shape_curl`](@ref)
+
+* [`function_value`](@ref)
+* [`function_gradient`](@ref)
+* [`function_symmetric_gradient`](@ref)
+* [`function_divergence`](@ref)
+* [`function_curl`](@ref)
+* [`spatial_coordinate`](@ref)
+"""
+InterfaceValues
+
+struct InterfaceValues{FVA, FVB} <: AbstractValues
+    face_values_a::FVA
+    face_values_b::FVB
+    interface_transformation::InterfaceTransformation
+end
+function InterfaceValues(quad_rule_a::FaceQuadratureRule, func_interpol_a::Interpolation,
+    geom_interpol_a::Interpolation = func_interpol_a; quad_rule_b::FaceQuadratureRule = deepcopy(quad_rule_a),
+    func_interpol_b::Interpolation = func_interpol_a, geom_interpol_b::Interpolation = func_interpol_b)
+    face_values_a = FaceValues(quad_rule_a, func_interpol_a, geom_interpol_a)
+    face_values_b = FaceValues(quad_rule_b, func_interpol_b, geom_interpol_b)
+    return InterfaceValues{typeof(face_values_a), typeof(face_values_b)}(face_values_a, face_values_b, InterfaceTransformation())
+end
+
+"""
+    reinit!(iv::InterfaceValues, face_a::FaceIndex, face_b::FaceIndex, cell_a_coords::AbstractVector{Vec{dim, T}}, cell_b_coords::AbstractVector{Vec{dim, T}}, grid::AbstractGrid) where {dim, T}
+
+Update the [`FaceValues`](@ref) in the interface (A and B) using their corresponding cell coordinates and [`FaceIndex`](@ref). This involved recalculating the transformation matrix [`transform_interface_point`](@ref)
+and mutating element B's quadrature points and its [`FaceValues`](@ref) `M, N, dMdξ, dNdξ`.
+"""
+function reinit!(iv::InterfaceValues, face_a::FaceIndex, face_b::FaceIndex, cell_a_coords::AbstractVector{Vec{dim, T}}, cell_b_coords::AbstractVector{Vec{dim, T}}, grid::AbstractGrid) where {dim, T}
+    reinit!(iv.face_values_a, cell_a_coords, face_a[2])
+    iv.face_values_b.current_face[] = face_b[2]
+    update_interface_transformation!(iv.interface_transformation, grid, face_a, face_b)
+    quad_points_a = getpoints(iv.face_values_a.qr, face_a[2])
+    quad_points_b = getpoints(iv.face_values_b.qr, face_b[2])
+    transform_interface_points!(quad_points_b, iv, quad_points_a, grid, face_a, face_b)
+    @boundscheck checkface(iv.face_values_b, face_b[2])
+    # This is the bottleneck, cache it?
+    for (idx, qp) in pairs(quad_points_b)
+        gradients_and_values = shape_gradients_and_values(iv.face_values_b.func_interp, qp)
+        iv.face_values_b.dNdξ[:,idx, face_b[2]] .= getindex.(gradients_and_values,1)
+        iv.face_values_b.N[:, idx, face_b[2]] .= getindex.(gradients_and_values,2)
```

Comment:
Seem easier to use a loop instead of creating the temporary svectors? Then you can also remove the new functions `shape_gradients_and_values` etc.